### PR TITLE
[structural binding][9/N] Replace std::tie with structural binding

### PR DESCRIPTION
**Pull request by cyyever**
Follows  #130544

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: aten/src/ATen/native/cpu/FusedAdamKernel.cpp [hunk 1]
```diff
@@ -233,13 +224,11 @@ typename std::enable_if<
     param_vec = param_vec + Vec(scalar_t(-step_size)) * exp_avg_vec / denom_vec;
     param_vec.store(param_ptr + d);
   }
-  scalar_t grad_val_to_store;
   for (; d < size; d++) {
     scalar_t grad_val = grad_ptr[d];
     if (grad_scale_ptr) {
       grad_val = grad_ptr[d] / scalar_t(*grad_scale_ptr);
```
Comment (janeyx99) [inline aten/src/ATen/native/cpu/FusedAdamKernel.cpp:230]: huh now i'm confused why we ever needed the two lines before this at all.

**Review by janeyx99 — APPROVED** [review]
looks good, please make sure the CPU tests run to completion before landing (to verify the fused optim changes)

File: aten/src/ATen/native/cpu/FusedAdamKernel.cpp (hunk 1 above)
Comment (cyyever) [reply aten/src/ATen/native/cpu/FusedAdamKernel.cpp:230]: Forgot to remove

**Comment by janeyx99** [comment]
Also, why did you add ciflow/binaries label? This change shouldn't affect the binaries (if it builds fine for the other tests).

**Comment by cyyever** [comment]
> Also, why did you add ciflow/binaries label? This change shouldn't affect the binaries (if it builds fine for the other tests).

Because it changed the fused optimisers a lot in a not obvious way, I wanted to test more.

**Pull request closed without merging**